src/pudl/transform/ferc1.py

```diff
@@ -717,63 +715,69 @@ def drop_null_data_rows(self, df: pd.DataFrame) -> pd.DataFrame:
     """Transformer class for the plants_steam_ferc1 table."""
 
-    def execute(
+    table_id: Ferc1TableId = Ferc1TableId.PLANTS_STEAM_FERC1
+
+    def transform(
         self,
         raw_dbf: pd.DataFrame,
         raw_xbrl_instant: pd.DataFrame,
         raw_xbrl_duration: pd.DataFrame,
         transformed_fuel: pd.DataFrame,
     ):
         """Perform table transformations for the plants_steam_ferc1 table."""
-        plants_steam_combo = (
-            self.pre_concat_clean_and_concat_dbf_xbrl(
-                raw_dbf, raw_xbrl_instant, raw_xbrl_duration
-            )
-            .pipe(self.simplify_strings_for_table)
-            .pipe(self.clean_strings_for_table)
-            .pipe(self.oob_to_nan_for_table)
-            .pipe(self.convert_units_in_table)
-            .pipe(self.convert_table_dtypes)
-            .pipe(plants_steam_assign_plant_ids, ferc1_fuel_df=transformed_fuel)
-        )
-        plants_steam_validate_ids(plants_steam_combo)
-        return plants_steam_combo
-
-    def process_dbf(self, raw_dbf):
-        """Modifications of the dbf plants_steam_ferc1 table before concat w/ xbrl."""
-        plants_steam_dbf = self.rename_columns(
-            raw_table=raw_dbf,
-            source="dbf",
-        ).pipe(self.assign_record_id, source_ferc1=Ferc1Source.DBF)
-        return plants_steam_dbf
-
-    def process_xbrl(self, raw_xbrl_instant, raw_xbrl_duration):
-        """Modifications of the xbrl plants_steam_ferc1 table before concat w/ xbrl."""
-        plants_steam_xbrl = (
-            self.merge_instant_and_duration_tables(
-                duration=raw_xbrl_duration,
-                instant=raw_xbrl_instant,
+        self.plants_steam = (
+            self.concat_dbf_xbrl(
+                raw_dbf=raw_dbf,
+                raw_xbrl_instant=raw_xbrl_instant,
+                raw_xbrl_duration=raw_xbrl_duration,
             )
+            .pipe(self.normalize_strings_multicol, params=self.params.normalize_strings)
+            .pipe(self.nullify_outliers_multicol, params=self.params.nullify_outliers)
             .pipe(
-                self.rename_columns,
-                source="xbrl",
+                self.categorize_strings_multicol, params=self.params.categorize_strings
             )
-            .pipe(self.add_report_year_column)
-            .pipe(
-                self.assign_record_id,
-                source_ferc1=Ferc1Source.XBRL,
+            .pipe(self.convert_units_multicol, params=self.params.convert_units)
+            .pipe(pudl.helpers.convert_cols_dtypes, data_source="eia")
+            .pipe(self.drop_null_data_rows)
+        )
+        self.plants_steam = self.plants_steam.pipe(
+            plants_steam_assign_plant_ids, ferc1_fuel_df=transformed_fuel
+        ).pipe(self.enforce_schema)
+        plants_steam_validate_ids(self.plants_steam)
+        return self.plants_steam
+
+    def drop_null_data_rows(self, df):
```

Comment:
I was trying to think about how to parameterize it... and haven't come up with a good idea yet. I think it's hard because we want to be flexible bc dropping things should look differently for different tables and will almost always involve masks, which feel hard to parameterize cleanly. but very open to suggestions here.

Reply:
this has been removed and generalized/parameterized.